src/internal/functions/Remove-AzOpsDeployment.ps1

```diff
@@ -0,0 +1,164 @@
+            Path where the ARM templates can be found.
+        .PARAMETER StatePath
+            The root folder under which to find the resource json.
+        .PARAMETER WhatIf
+            If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+        .EXAMPLE
+            > $AzOpsRemovalList | Select-Object $uniqueProperties -Unique | Remove-AzOpsDeployment
+            Remove all unique deployments provided from $AzOpsRemovalList
+    #>
+
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    param (
+        [Parameter(ValueFromPipelineByPropertyName = $true)]
+        [string]
+        $TemplateFilePath = (Get-PSFConfigValue -FullName 'AzOps.Core.MainTemplate'),
+
+        [string]
+        $StatePath = (Get-PSFConfigValue -FullName 'AzOps.Core.State')
+    )
+
+    process {
+        #Deployment Name
+        $fileItem = Get-Item -Path $TemplateFilePath
+        $deploymentName = $fileItem.BaseName -replace '\.json$' -replace ' ', '_'
+        $deploymentName = "AzOps-RemoveResource-$deploymentName"
+        
+        Write-PSFMessage -String 'Remove-AzOpsDeployment.Processing' -StringValues $deploymentName, $TemplateFilePath -Target $TemplateFilePath
+
+        #region Parse Content
+        $templateContent = Get-Content $TemplateFilePath | ConvertFrom-Json -AsHashtable
+        #endregion
+
+        #region Validate it is AzOpsgenerated template
+        if($templateContent.metadata._generator.name -eq "AzOps")
+        {
+            Write-PSFMessage -Level Verbose -Message 'Remove-AzOpsDeployment.Metadata.Success' -StringValues $TemplateFilePath -Target $TemplateFilePath
+        }
+        else
+        {
+            Write-PSFMessage -Level Error -Message 'Remove-AzOpsDeployment.Metadata.Failed' -StringValues $TemplateFilePath -Target $TemplateFilePath
+            return
+        }
+        #endregion Validate it is AzOpsgenerated template
+
+        #region Resolve Scope
+        try {
+            $scopeObject = New-AzOpsScope -Path $TemplateFilePath -StatePath $StatePath -ErrorAction Stop -WhatIf:$false
+        }
+        catch {
+            Write-PSFMessage -Level Warning -String 'Remove-AzOpsDeployment.Scope.Failed' -Target $TemplateFilePath -StringValues $TemplateFilePath -ErrorRecord $_
+            return
+        }
+        if (-not $scopeObject) {
+            Write-PSFMessage -Level Warning -String 'Remove-AzOpsDeployment.Scope.Empty' -Target $TemplateFilePath -StringValues $TemplateFilePath
+            return
+        }
+        #endregion Resolve Scope
+
+        #region SetContext
+        Set-AzOpsContext -ScopeObject $scopeObject
+        #endregion SetContext
+
+        #GetContext
+        $context = Get-AzContext
+
+        #region PolicyAssignment
+        if($scopeObject.Resource -eq "policyAssignments"){
+
+            #Validate
+            $roleAssignmentPermissionCheck = $false
```

Comment:
To avoid duplicating code, can we create a generic function outside of this function to validate the permissions?

Reply:
Agree, Generic function has been added.